dimuon/terraform-provider-ec
# Migrate `ec_deployment`

## Description


**`ec_deployment` resource migration**

*Limitations.*
State upgrade is not currently implemented.

Few acceptance tests fail. Please see the section below.

## Related Issues






## Motivation and Context


## How Has This Been Tested?

* unit tests (majority of the relevant unit tests were migrated)
* acceptance tests
* manual testing

There are still few acceptance tests that fail:
- there are 4 acceptance tests (their names mention version 0.4.1) that verify migration of other data sources and resources from 0.4.1 to 0.6.0. To fix them we need implement a state upgrade for `ec_deployment`.
- `TestAccDeploymentExtension_pluginDownload` fails on 0.4.1 with the same error ("Custom plugin is not allowed at your subscription level.")
- `TestAccDeploymentExtension_bundleFile` fails quite often on both 0.4.1 and the branch with the "failed to upload file" error.
- `TestAccDeployment_withExtension` fails quite often with the same error as `TestAccDeploymentExtension_bundleFile`.





## Types of Changes


- [ ] Refactoring (improves code quality but has no user-facing effect)

## Readiness Checklist
